src/Blumchen/DependencyInjection/WorkerOptionsBuilder.cs

```diff
@@ -0,0 +1,67 @@
+using Blumchen.Subscriber;
+using Blumchen.Subscriptions.Management;
+using Npgsql;
+using Npgsql.Replication;
+using Polly;
+
+namespace Blumchen.DependencyInjection;
+
+public record WorkerOptions(
+    ISubscriberOptions SubscriberOptions,
+    ResiliencePipeline OuterPipeline,
+    ResiliencePipeline InnerPipeline);
+
+public interface IWorkerOptionsBuilder
+{
+    IWorkerOptionsBuilder ResiliencyPipeline(ResiliencePipeline resiliencePipeline);
+    IWorkerOptionsBuilder Subscription(Func<OptionsBuilder, OptionsBuilder>? builder);
+    WorkerOptions Build();
+    IWorkerOptionsBuilder EnableSubscriptionAutoHeal();
+}
+
+internal sealed class WorkerOptionsBuilder: IWorkerOptionsBuilder
+{
+    private ResiliencePipeline? _outerPipeline = default;
+    private Func<string, string, ResiliencePipeline>? _innerPipelineFn = default;
+    private Func<OptionsBuilder, OptionsBuilder>? _builder;
+
+    public IWorkerOptionsBuilder ResiliencyPipeline(ResiliencePipeline resiliencePipeline)
+    {
+        _outerPipeline = resiliencePipeline;
+        return this;
+    }public IWorkerOptionsBuilder Subscription(Func<OptionsBuilder, OptionsBuilder>? builder)
+    {
+        _builder = builder;
+        return this; 
+    }
+
+    public WorkerOptions Build()
+    {
+        ArgumentNullException.ThrowIfNull(_outerPipeline);
```

Comment:
**SUGGESTION:** We should not force users to set a resiliency pipeline. In most cases, they don't know what the best option is when they start to use the tool and approach. So it's best to set up the safe defaults for them, so they can diverge when the default choice is not enough. Same goes for other settings. Of course, some has to be defined like handlers etc. but we should strive to provide as much safe defaults as possible. 

We can inject the service provider and by convention try to get it from DI.

Reply:
I’ll provide a UseDefault() approach where possible in order to minimise initial friction